### PR TITLE
CAS Security cleanup

### PR DESCRIPTION
**Pull request by jkotas**
- Delete most CAS support under mscorlib\src\System\Security. Introduce temporary dummy stubs for a few types that are referenced in many places, to be cleaned up later. https://github.com/dotnet/coreclr/pull/9355/files#diff-7f73d6dbdafd14ee136ff60b85276236
- Delete enough on the unmanaged side to break ties with the removed types on the managed side. There is still quite a bit left though.

Contributes to https://github.com/dotnet/coreclr/issues/9321

### CODE REVIEW

**Comment by jkotas** [comment]
cc @danmosemsft

**Comment by jkotas** [comment]
mscorlib.shared.sources.props shows up as all diff because of https://github.com/dotnet/coreclr/issues/9353

**Comment by jkotas** [comment]
I have run coreclr and corefx tests on this.

@danmosemsft @gkhanna79 Could you please take a cursory look?

File: src/mscorlib/src/System/Reflection/Assembly.cs [hunk 1]
```diff
@@ -1268,11 +1239,11 @@ internal bool IsSafeForReflection()
                                              IntPtr pPrivHostBinder,
                                              bool throwOnFileNotFound,
                                              bool forIntrospection,
-                                             bool suppressSecurityChecks, IntPtr ptrLoadContextBinder = default(IntPtr))
+                                             IntPtr ptrLoadContextBinder = default(IntPtr))
         {
             return _nLoad(fileName, codeBase, assemblySecurity, locationHint, ref stackMark,
                 pPrivHostBinder,
-                throwOnFileNotFound, forIntrospection, suppressSecurityChecks, ptrLoadContextBinder);
+                throwOnFileNotFound, forIntrospection, true, ptrLoadContextBinder);
```
Comment (danmosemsft) [inline src/mscorlib/src/System/Reflection/Assembly.cs]: >true [](start = 55, length = 4)

nit, it is nice to prefix bools 
like /* suppressSecurityChecks = */ true
but I see this is not widely done.

**Comment by danmosemsft** [comment]
LGTM in a quick look. I hope we have enough test coverage of all the parameter validation.

**Pull request merged**